staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/peer_aggregated_handler_test.go

```diff
@@ -0,0 +1,423 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aggregated_test
```

Comment:
Sorry to ask but could we add unit test function that specifically unit test `fetchFromCache`, `mergeResources` and the other non-trivial functions declared in `peer_aggregated_handler.go` in isolation?  I'd find that helpful in confirming my understanding of those functions.

Reply:
tests for mergeResources are already there, added a new `TestFetchFromCache()` for `fetchFromCache `